src/waterfall/polarplot.cpp

```diff
@@ -74,6 +74,7 @@ void PolarPlot::draw(const QVector<double>& points, float angle, float initPoint
     static float angleResolution = gradianToDegree/2;
     float actualAngle = angle*gradianToRadian;
 
+    //TODO: Need a better way to deal with dynamic steps
```

Comment:
An idea: instead of adding angleGrad parameter to this function, the caller could just loop over the sweep angle, and call `draw(data, angle++)`. This will sacrifice performance in calls to draw, but we have to meet that performance requirement anyway when the step resolution == 1, so I don't think that sacrifice matters.

Reply:
TODO updated.